src/NHibernate.Test/NHSpecificTest/NH2319/MapFixture.cs

```diff
@@ -0,0 +1,286 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using NHibernate.Cfg;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Collection;
+using NHibernate.Engine.Query;
+using NHibernate.Mapping.ByCode;
+using NHibernate.Util;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.NH2319
+{
+	[TestFixture]
+	public class MapFixture : TestCaseMappingByCode
+	{
+		private Guid _parent1Id;
+		private Guid _child1Id;
+		private Guid _parent2Id;
+		private Guid _child3Id;
+
+		[Test]
+		public void ShouldBeAbleToFindChildrenByName()
+		{
+			FindChildrenByName(_parent1Id, _child1Id);
+		}
+
+		private void FindChildrenByName(Guid parentId, Guid childId)
+		{
+			using (var session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				var parent = session.Get<Parent>(parentId);
+
+				Assert.That(parent, Is.Not.Null);
+
+				var filtered = parent.ChildrenMap.Values
+					.AsQueryable()
+					.Where(x => x.Name == "Jack")
+					.ToList();
+
+				Assert.That(filtered, Has.Count.EqualTo(1));
+				Assert.That(filtered[0].Id, Is.EqualTo(childId));
+			}
+		}
+
+		[Test]
+		public void ShouldBeAbleToPerformComplexFiltering()
+		{
+			using (var session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				var parent = session.Get<Parent>(_parent1Id);
+
+				Assert.NotNull(parent);
+
+				var filtered = parent.ChildrenMap.Values
+					.AsQueryable()
+					.Where(x => x.Name == "Piter")
+					.SelectMany(x => x.GrandChildren)
```

Comment:
Not a map here. Supporting maps inside lambda would be another feature. (Currently already doable by pretending the map is a `ICollection<TValue>` with a cast, but a bit hackish.)